### PR TITLE
Add more examples

### PR DESCRIPTION
**Pull request by Fleid**
Gives more visual examples on how to use the function
Not sure if queries 2 and 3 should be here or somewhere else, since they don't use GetArrayElement, but from a user perspective it sounds logical to flow from GetArrayElement to GetArrayElements here

### CODE REVIEW

**Comment by PRMerger17** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 1]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+|DeviceId|CustomerSensorValue|
+|---|---|
+|631|0|
+|631|1|
+|631|0|
+|123|0|
+|123|1|
+|123|1|
+
+From there it is straightforward to aggregate the content if necessary:
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
From there, you can easily aggregate the content if necessary:
```

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 2]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+```
+
+Will return
+
+|DeviceId|Temperature|Humidity|CustomSensorArray|FirstCustomSensorValue|SecondCustomSensorValue|
+|---|---|---|---|---|---|
+|631|81|69|0,1,0|0|1|
+|123|80|70|1,1,0|1|1|
+
+In order to unfold the array, [CROSS APPLY](apply-azure-stream-analytics.md) can be leveraged:
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
Use [CROSS APPLY](apply-azure-stream-analytics.md) to unfold the array:
```

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 3]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+
+At the individual record level, we have 2 properties with different [types](https://docs.microsoft.com/en-us/stream-analytics-query/data-types-azure-stream-analytics). `DeviceId` is of type **nvarchar(max)**, `SensorReadings` is of type **record** (object). [GetType](https://docs.microsoft.com/en-us/stream-analytics-query/gettype-azure-stream-analytics) can be used to determine the type when necessary.
+
+`SensorReadings` has 3 properties, 2 of type **bigint**: `Temperature` and `Humidity`, and finally `CustomSensor` of type **array** (of **bigint**). If this array was in turn more complex (itself containing records or arrays), a combination of [GetArrayElements](https://docs.microsoft.com/en-us/stream-analytics-query/getarrayelements-azure-stream-analytics]) (plural) and [GetRecordPropertyValue](https://docs.microsoft.com/en-us/stream-analytics-query/getrecordpropertyvalue-azure-stream-analytics) could be leveraged.
+
+#### Queries
+
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]: add an intro to this query

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 4]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+```SQL  
+SELECT   
+	i.DeviceId,
+	CustomerSensorValue.ArrayValue AS CustomerSensorValue
+FROM input AS i
+CROSS APPLY GetArrayElements(i.SensorReadings.CustomSensor) AS CustomerSensorValue
+``` 
+
+Will return
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
Returns the following output:
```

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 5]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+	i.DeviceId,
+	i.SensorReadings.Temperature,
+	i.SensorReadings.Humidity,
+	i.SensorReadings.CustomSensor as CustomSensorArray,
+	GetArrayElement(i.SensorReadings.CustomSensor,0) AS FirstCustomSensorValue,
+  GetArrayElement(i.SensorReadings.CustomSensor,1) AS SecondCustomSensorValue
+FROM input i
+```
+
+Will return
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
Returns the following output:
```

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 6]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+    {
+        "Temperature" : 81,
+        "Humidity" : 69,
+        "CustomSensor": [0,1,0]
+    }
+}
+]
+```
+
+The sample dataset above is an array of 2 records. When used as [local input](https://docs.microsoft.com/en-us/azure/stream-analytics/visual-studio-code-local-run) in a json file, the top level array will be interpreted for the generation of rows / events by Azure Stream Analytics. There is no need to take it into consideration in the query syntax.
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
The sample dataset above is an array of two records. When used as [local input](https://docs.microsoft.com/en-us/azure/stream-analytics/visual-studio-code-local-run) in a json file, the top level array is interpreted for the generation of rows/events by Azure Stream Analytics. There is no need to take it into consideration in the query syntax.
```

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 7]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+
+At the individual record level, we have 2 properties with different [types](https://docs.microsoft.com/en-us/stream-analytics-query/data-types-azure-stream-analytics). `DeviceId` is of type **nvarchar(max)**, `SensorReadings` is of type **record** (object). [GetType](https://docs.microsoft.com/en-us/stream-analytics-query/gettype-azure-stream-analytics) can be used to determine the type when necessary.
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
At the individual record level, there are two properties with different [types](https://docs.microsoft.com/en-us/stream-analytics-query/data-types-azure-stream-analytics). `DeviceId` is of type **nvarchar(max)**, `SensorReadings` is of type **record** (object). [GetType](https://docs.microsoft.com/en-us/stream-analytics-query/gettype-azure-stream-analytics) can be used to determine the type when necessary.
```

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md [hunk 8]
```diff
@@ -37,15 +37,96 @@ GetArrayElement ( array_expression, bigint_expression )
...
+
+`SensorReadings` has 3 properties, 2 of type **bigint**: `Temperature` and `Humidity`, and finally `CustomSensor` of type **array** (of **bigint**). If this array was in turn more complex (itself containing records or arrays), a combination of [GetArrayElements](https://docs.microsoft.com/en-us/stream-analytics-query/getarrayelements-azure-stream-analytics]) (plural) and [GetRecordPropertyValue](https://docs.microsoft.com/en-us/stream-analytics-query/getrecordpropertyvalue-azure-stream-analytics) could be leveraged.
```
Comment (mamccrea) [inline stream-analytics-docs/getarrayelement-azure-stream-analytics.md]:
```suggestion
`SensorReadings` has three properties: two are of type **bigint**: `Temperature` and `Humidity`, and `CustomSensor` is of type **array** (of **bigint**). If this array was more complex (itself containing records or arrays), a combination of [GetArrayElements](https://docs.microsoft.com/en-us/stream-analytics-query/getarrayelements-azure-stream-analytics]) (plural) and [GetRecordPropertyValue](https://docs.microsoft.com/en-us/stream-analytics-query/getrecordpropertyvalue-azure-stream-analytics) could be used.
```

**Comment by PRMerger13** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger13** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger13** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger13** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger14** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger14** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger14** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger15** [comment]
@Fleid : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: stream-analytics-docs/getarrayelement-azure-stream-analytics.md (hunk 3 above)
Comment (Fleid) [reply stream-analytics-docs/getarrayelement-azure-stream-analytics.md]: Addressed in [98e4ed4](https://github.com/MicrosoftDocs/azure-reference-other/pull/104/commits/98e4ed4586510f595ac94d76ed0ec9711b42ca79)

**Comment by mamccrea** [comment]
#sign-off

**Comment by PRMerger6** [comment]
@ktoliver : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Pull request merged**